In [48]:
import pandas as pd


In [49]:
# pip install pyarrow #para convertir a parquet el dataframe final
import pyarrow

In [50]:
#carga del arhivo movies.

df = pd.read_csv("movies_dataset.csv")
df.head()

C:\Users\fmgc1\AppData\Local\Temp\ipykernel_13108\215091187.py:3: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("movies_dataset.csv")


adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ... release_date  \
0  Led by Woody, Andy's toys live happily in his ...  ...   1995-10-30   
1  When siblings Judy and Peter discover an encha...  ...   1995-12-15   
2  A family wedding reignites the ancient feud be...  ...   1995-12-22   
3  Cheated on, mistreated and stepped on, the wom...  ...   1995-12-22   
4  Just when George Banks has recovered from his ...  ...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

     status                                            tagline  \
0  Released                                                NaN   
1  Released          Roll the dice and unleash the excitement!   
2  Released  Still Yelling. Still Fighting. Still Ready for...   
3  Released  Friends are the people who let you be yourself...   
4  Released  Just When His World Is Back To Normal... He's ...   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 24 columns]

De un primer analisis rapido y consultando el diccionario de datos pudimos identificar algunas columnas que consideramos que no aportarian informacion util al algoritmo de recomendacion. Por ejemplo la columna homepage contiene las direcciones web de las paginas de las peliculas, la columna overview contiene una reseña del contenido de la pelicula.

In [51]:
#Eliminamos las columnas que consideramos que no aportarian informacion util
# El resto se hara en el apartado de TRANSFORMACIONES mas abajo.
df = df.drop(columns=['status', 'tagline'])
df.head()

adult                              belongs_to_collection    budget  \
0  False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1  False                                                NaN  65000000   
2  False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3  False                                                NaN  16000000   
4  False  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
4                     [{'id': 35, 'name': 'Comedy'}]   

                               homepage     id    imdb_id original_language  \
0  http://toystory.disney.com/toy-story    862  tt0114709                en   
1                                   NaN   8844  tt0113497                en   
2                                   NaN  15602  tt0113228                en   
3                                   NaN  31357  tt0114885                en   
4                                   NaN  11862  tt0113041                en   

                original_title  \
0                    Toy Story   
1                      Jumanji   
2             Grumpier Old Men   
3            Waiting to Exhale   
4  Father of the Bride Part II   

                                            overview  ...  \
0  Led by Woody, Andy's toys live happily in his ...  ...   
1  When siblings Judy and Peter discover an encha...  ...   
2  A family wedding reignites the ancient feud be...  ...   
3  Cheated on, mistreated and stepped on, the wom...  ...   
4  Just when George Banks has recovered from his ...  ...   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue runtime                                   spoken_languages  \
0  373554033.0    81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0   104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0   101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0   127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0   106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

                         title  video vote_average vote_count  
0                    Toy Story  False          7.7     5415.0  
1                      Jumanji  False          6.9     2413.0  
2             Grumpier Old Men  False          6.5       92.0  
3            Waiting to Exhale  False          6.1       34.0  
4  Father of the Bride Part II  False          5.7      173.0  

[5 rows x 22 columns]

In [52]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 22 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   adult                  45466 non-null  object 
 1   belongs_to_collection  4494 non-null   object 
 2   budget                 45466 non-null  object 
 3   genres                 45466 non-null  object 
 4   homepage               7782 non-null   object 
 5   id                     45466 non-null  object 
 6   imdb_id                45449 non-null  object 
 7   original_language      45455 non-null  object 
 8   original_title         45466 non-null  object 
 9   overview               44512 non-null  object 
 10  popularity             45461 non-null  object 
 11  poster_path            45080 non-null  object 
 12  production_companies   45463 non-null  object 
 13  production_countries   45463 non-null  object 
 14  release_date           45379 non-null  object 
 15  re

Notamos que de de las 45466 filas totales la columna title contiene 6 valores nulos, entonces los analizamos

In [52]:
df[df['title'].isna()]

adult  \
19729                                              False   
19730                                 - Written by Ørnås   
29502                                              False   
29503   Rune Balot goes to a casino connected to the ...   
35586                                              False   
35587   Avalanche Sharks tells the story of a bikini ...   

                                   belongs_to_collection  \
19729                                                NaN   
19730                                           0.065736   
29502  {'id': 122661, 'name': 'Mardock Scramble Colle...   
29503                                           1.931659   
35586                                                NaN   
35587                                           2.185485   

                                 budget  \
19729                                 0   
19730  /ff9qCepilowshEtG2GYWwzt2bs4.jpg   
29502                                 0   
29503  /zV8bHuSL6WXoD6FWogP9j4x80bL.jpg   
35586                                 0   
35587  /zaSf5OG7V8X8gqFvly88zDdRm46.jpg   

                                                  genres  \
19729  [{'id': 28, 'name': 'Action'}, {'id': 53, 'nam...   
19730  [{'name': 'Carousel Productions', 'id': 11176}...   
29502  [{'id': 16, 'name': 'Animation'}, {'id': 878, ...   
29503  [{'name': 'Aniplex', 'id': 2883}, {'name': 'Go...   
35586  [{'id': 10770, 'name': 'TV Movie'}, {'id': 28,...   
35587  [{'name': 'Odyssey Media', 'id': 17161}, {'nam...   

                                                homepage          id  \
19729                                                NaN       82663   
19730  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...  1997-08-20   
29502                      http://m-scramble.jp/exhaust/      122662   
29503  [{'iso_3166_1': 'US', 'name': 'United States o...  2012-09-29   
35586                                                NaN      249260   
35587           [{'iso_3166_1': 'CA', 'name': 'Canada'}]  2014-01-01   

         imdb_id original_language                            original_title  \
19729  tt0113002                en                              Midnight Man   
19730          0             104.0  [{'iso_639_1': 'en', 'name': 'English'}]   
29502  tt2423504                ja                          マルドゥック・スクランブル 排気   
29503          0              68.0      [{'iso_639_1': 'ja', 'name': '日本語'}]   
35586  tt2622826                en                          Avalanche Sharks   
35587          0              82.0  [{'iso_639_1': 'en', 'name': 'English'}]   

                  popularity  ... production_companies production_countries  \
19729                    NaN  ...                  NaN                  NaN   
19730                    NaN  ...                False                  6.0   
29502                    NaN  ...                  NaN                  NaN   
29503                    NaN  ...                False                  7.0   
35586                    NaN  ...                  NaN                  NaN   
35587  Beware Of Frost Bites  ...                False                  4.3   

      release_date revenue  runtime  spoken_languages title video  \
19729          NaN     NaN      NaN               NaN   NaN   NaN   
19730            1     NaN      NaN               NaN   NaN   NaN   
29502          NaN     NaN      NaN               NaN   NaN   NaN   
29503           12     NaN      NaN               NaN   NaN   NaN   
35586          NaN     NaN      NaN               NaN   NaN   NaN   
35587           22     NaN      NaN               NaN   NaN   NaN   

      vote_average  vote_count  
19729          NaN         NaN  
19730          NaN         NaN  
29502          NaN         NaN  
29503          NaN         NaN  
35586          NaN         NaN  
35587          NaN         NaN  

[6 rows x 21 columns]

Notamos que podemos eliminar del dataset estas 6 filas

In [53]:
# elimino las filas cuyo valores en la columna title es NAN
df = df.dropna(subset=['title'])

Analizamos ahora la columna adult

In [6]:
df['adult'].value_counts()

adult
False    45451
True         9
Name: count, dtype: int64

Notamos 9 filas con valor TRUE las analizamos en detalle

In [55]:
df[df['adult'] == "True"]

adult belongs_to_collection  budget  \
19489  True                   NaN       0   
28701  True                   NaN       0   
31934  True                   NaN       0   
32113  True                   NaN       0   
39901  True                   NaN       0   
39902  True                   NaN       0   
40574  True                   NaN       0   
41009  True                   NaN  750000   
43090  True                   NaN       0   

                                                  genres  \
19489                     [{'id': 27, 'name': 'Horror'}]   
28701                      [{'id': 80, 'name': 'Crime'}]   
31934                     [{'id': 35, 'name': 'Comedy'}]   
32113  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   
39901  [{'id': 80, 'name': 'Crime'}, {'id': 27, 'name...   
39902  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...   
40574  [{'id': 80, 'name': 'Crime'}, {'id': 27, 'name...   
41009  [{'id': 53, 'name': 'Thriller'}, {'id': 80, 'n...   
43090  [{'id': 878, 'name': 'Science Fiction'}, {'id'...   

                                   homepage      id    imdb_id  \
19489                                   NaN    5422  tt0079642   
28701  http://www.imdb.com/title/tt0119931/  169158  tt0119931   
31934                                   NaN   44781  tt0322232   
32113             http://www.dietofsex.com/  324230  tt3094816   
39901                                   NaN   35731  tt1161951   
39902                                   NaN   55774  tt1153101   
40574                                   NaN   39922  tt0070696   
41009                                   NaN  376004  tt4044464   
43090                                   NaN  448290  tt2697688   

      original_language                       original_title popularity  ...  \
19489                it  Le notti erotiche dei morti viventi   1.420104  ...   
28701                en                             Standoff   1.349745  ...   
31934                cn                                發電悄嬌娃   0.001803  ...   
32113                es                          Diet of Sex   6.889832  ...   
39901                en           Amateur Porn Star Killer 2   0.587751  ...   
39902                en                             The Band   0.417214  ...   
40574                da                              Dværgen   1.182635  ...   
41009                en                           Adulterers    1.13377  ...   
43090                en                           Half -Life    0.28268  ...   

                                    production_companies  \
19489             [{'name': 'Stefano Film', 'id': 6117}]   
28701         [{'name': 'Trimark Pictures', 'id': 4063}]   
31934                                                 []   
32113              [{'name': 'KeyAtomics', 'id': 46055}]   
39901         [{'name': 'Alter Ego Cinema', 'id': 5639}]   
39902                                                 []   
40574           [{'name': 'Scandica Film', 'id': 48812}]   
41009                                                 []   
43090  [{'name': 'Bad Robot', 'id': 11461}, {'name': ...   

                                    production_countries release_date revenue  \
19489            [{'iso_3166_1': 'IT', 'name': 'Italy'}]   1980-11-18     0.0   
28701  [{'iso_3166_1': 'US', 'name': 'United States o...   1998-01-11     0.0   
31934        [{'iso_3166_1': 'HK', 'name': 'Hong Kong'}]   2001-04-26     0.0   
32113            [{'iso_3166_1': 'ES', 'name': 'Spain'}]   2014-02-14     0.0   
39901  [{'iso_3166_1': 'US', 'name': 'United States o...   2008-05-13     0.0   
39902        [{'iso_3166_1': 'AU', 'name': 'Australia'}]   2009-11-17     0.0   
40574  [{'iso_3166_1': 'DK', 'name': 'Denmark'}, {'is...   1973-12-20     0.0   
41009                                                 []   2016-01-05     0.0   
43090  [{'iso_3166_1': 'US', 'name': 'United States o...          NaN     0.0   

       runtime                            spoken_languages  \
19489    112.0   [

se decide hacer un recomendador "family friendly" por lo que nos quedamos solo con los titulos que no son para adultos

In [54]:
df = df[df['adult'] == "False"]

Una vez echo esto la columna adult queda con todos los valores "False" por lo que se puede quitar esta columna

In [32]:
#comprobamos
df['adult'].value_counts()

adult
False    45451
Name: count, dtype: int64

In [24]:
#Se hara mas adelante en el apartado TRANSFORMACIONES
#eliminamos la columna
#df = df.drop('adult', axis=1)

## Transformaciones

1_ Eliminar las columnas innecesarias para el desarrollo de la API, como "video", "imdb_id", "adult", "original_title", "vote_count", "poster_path" y "homepage".

2_ Rellenar con 0 los registros con valores nulos en las columnas "revenue" y "budget".

3_ Eliminar los registros que contengan valores nulos en la columna "release date".

4_ Asegurarse de que la columna "release_year" esté en formato AAAA-MM-DD y generar una nueva columna que contenga solo el año de estreno.

5_ Crear una nueva columna "return" que extraiga el retorno de inversión de cada registro, dividiendo la columna "revenue" entre la columna "budget".

6_ Desanidar los datos que se encuentran en formato de listas de diccionarios para poder unirlos nuevamente a la base de datos de manera separada.



In [55]:
#1 Eliminamos columnas inecesarias:
df = df.drop(columns=['video', 'imdb_id','adult', 'original_title', 'poster_path','homepage'])
df.head()


belongs_to_collection    budget  \
0  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1                                                NaN  65000000   
2  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
3                                                NaN  16000000   
4  {'id': 96871, 'name': 'Father of the Bride Col...         0   

                                              genres     id original_language  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...    862                en   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   8844                en   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...  15602                en   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...  31357                en   
4                     [{'id': 35, 'name': 'Comedy'}]  11862                en   

                                            overview popularity  \
0  Led by Woody, Andy's toys live happily in his ...  21.946943   
1  When siblings Judy and Peter discover an encha...  17.015539   
2  A family wedding reignites the ancient feud be...    11.7129   
3  Cheated on, mistreated and stepped on, the wom...   3.859495   
4  Just when George Banks has recovered from his ...   8.387519   

                                production_companies  \
0     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3  [{'name': 'Twentieth Century Fox Film Corporat...   
4  [{'name': 'Sandollar Productions', 'id': 5842}...   

                                production_countries release_date  \
0  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4  [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   

       revenue  runtime                                   spoken_languages  \
0  373554033.0     81.0           [{'iso_639_1': 'en', 'name': 'English'}]   
1  262797249.0    104.0  [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2          0.0    101.0           [{'iso_639_1': 'en', 'name': 'English'}]   
3   81452156.0    127.0           [{'iso_639_1': 'en', 'name': 'English'}]   
4   76578911.0    106.0           [{'iso_639_1': 'en', 'name': 'English'}]   

                         title  vote_average  vote_count  
0                    Toy Story           7.7      5415.0  
1                      Jumanji           6.9      2413.0  
2             Grumpier Old Men           6.5        92.0  
3            Waiting to Exhale           6.1        34.0  
4  Father of the Bride Part II           5.7       173.0

In [56]:
#2_ Rellenar con 0 los registros con valores nulos en las columnas "revenue" y "budget".

df["budget"] = df["budget"].fillna(0)
df["revenue"] = df["revenue"].fillna(0)

In [57]:
#3_ Eliminar los registros que contengan valores nulos en la columna "release date".
df.dropna(subset=['release_date'], inplace=True)

In [58]:
#4_ Asegurarse de que la columna "release_year" esté en formato AAAA-MM-DD y generar una nueva columna que contenga solo el año de estreno.
df['release_date'] = pd.to_datetime(df['release_date'], errors='coerce')
df['release_year'] = df['release_date'].dt.year

In [59]:
#comprobamos la nueva columna con solo el año de estreno
df['release_year']

0        1995
1        1995
2        1995
3        1995
4        1995
         ... 
45460    1991
45462    2011
45463    2003
45464    1917
45465    2017
Name: release_year, Length: 45368, dtype: int32

In [60]:
# 5_ Crear una nueva columna "return" que extraiga el retorno de inversión de cada registro, 
# dividiendo la columna "revenue" entre la columna "budget".

#primero pasamos a numeric la columna Budget que figura como String
df['budget'] = pd.to_numeric(df['budget'], errors='coerce')

#creamos la nueva columna 
df['return'] = df['revenue'] / df['budget']
df['return']


0        12.451801
1         4.043035
2              NaN
3         5.090760
4              inf
           ...    
45460          NaN
45462          NaN
45463          NaN
45464          NaN
45465          NaN
Name: return, Length: 45368, dtype: float64

In [61]:
# 6_ Desanidar los datos que se encuentran en formato de listas de diccionarios 
# para poder unirlos nuevamente a la base de datos de manera separada.

#creamos la funcion de convertir a diccionario el texto
def str_to_dict(text):
  try:
    return eval(text)
  except:
    return {}

#aplicamos la funcion a la columna
df['belongs_to_collection'] = df['belongs_to_collection'].apply(str_to_dict)


In [62]:
#una vez como diccionario podemos aplicar pd.json_normalize para desanidar
df = pd.concat([pd.json_normalize(df['belongs_to_collection']), df.drop(['belongs_to_collection'], axis=1)], axis=1)

In [15]:
df

id                            name  \
0       10194.0            Toy Story Collection   
1           NaN                             NaN   
2      119050.0       Grumpy Old Men Collection   
3           NaN                             NaN   
4       96871.0  Father of the Bride Collection   
...         ...                             ...   
45460       NaN                             NaN   
45462       NaN                             NaN   
45463       NaN                             NaN   
45464       NaN                             NaN   
45465       NaN                             NaN   

                            poster_path                     backdrop_path  \
0      /7G9915LfUQ2lVfwMEEhDsn3kT4B.jpg  /9FBwqcd9IRruEDUrTdcaafOMKUq.jpg   
1                                   NaN                               NaN   
2      /nLvUdqgPgm3F85NMCii9gVFUcet.jpg  /hypTnLot2z8wpFS7qwsQHW1uV8u.jpg   
3                                   NaN                               NaN   
4      /nts4iOmNnq7GNicycMJ9pSAn204.jpg  /7qwE57OVZmMJChBpLEbJEmzUydk.jpg   
...                                 ...                               ...   
45460                               NaN                               NaN   
45462                               NaN                               NaN   
45463                               NaN                               NaN   
45464                               NaN                               NaN   
45465                               NaN                               NaN   

           budget                                             genres      id  \
0      30000000.0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     862   
1      65000000.0  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    8844   
2             0.0  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   15602   
3      16000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   31357   
4             0.0                     [{'id': 35, 'name': 'Comedy'}]   11862   
...           ...                                                ...     ...   
45460         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   30840   
45462         0.0                      [{'id': 18, 'name': 'Drama'}]  111109   
45463         0.0  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   67758   
45464         0.0                                                 []  227506   
45465         0.0                                                 []  461257   

      original_language                                           overview  \
0                    en  Led by Woody, Andy's toys live happily in his ...   
1                    en  When siblings Judy and Peter discover an encha...   
2                    en  A family wedding reignites the ancient feud be...   
3                    en  Cheated on, mistreated and stepped on, the wom...   
4                    en  Just when George Banks has recovered from his ...   
...                 ...                                                ...   
45460                en  Yet another version of the classic epic, with ...   
45462                tl  An artist struggles to finish his work while a...   
45463                en  When one of her hits goes wrong, a professiona...   
45464                en  In a small town live two brothers, one a minis...   
45465                en  50 years after decriminalisation of homosexual...   

      popularity  ...                               production_countries  \
0      21.946943  ...  [{'iso_3166_1': 'US', 'name': 'United States o...   
1      17.015539  ...  [{'iso_3166_1': 'US', 'name': 'United States o...   
2        11.7129  ...  [{'iso_3166_1': 'US', 'name': 'United States o...   
3       3.859495  ...  [{'iso_3166_1': 'US', 'name': 'United States o...   
4       8.387519  ...  [{'iso_3166_1': 'US', 'name': 'United States o...   
...          ...  ...                                                ...   
45460   5.683753  ...  [{'iso_3166_1': 'CA', 

Podemos notar que se desanido la columna Belongs_to_collection generandose 4 nuevas columnas.

In [63]:
#de esta nuevas columnas solo nos quedamos con name por el momento
#Elimino la primer columna
df=df.iloc[:, 1:]
#elimino las otras 2
df = df.drop(columns=['poster_path', 'backdrop_path'])

In [64]:
df

name      budget  \
0                Toy Story Collection  30000000.0   
1                                 NaN  65000000.0   
2           Grumpy Old Men Collection         0.0   
3                                 NaN  16000000.0   
4      Father of the Bride Collection         0.0   
...                               ...         ...   
45460                             NaN         0.0   
45462                             NaN         0.0   
45463                             NaN         0.0   
45464                             NaN         0.0   
45465                             NaN         0.0   

                                                  genres      id  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     862   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    8844   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   15602   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...   31357   
4                         [{'id': 35, 'name': 'Comedy'}]   11862   
...                                                  ...     ...   
45460  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   30840   
45462                      [{'id': 18, 'name': 'Drama'}]  111109   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...   67758   
45464                                                 []  227506   
45465                                                 []  461257   

      original_language                                           overview  \
0                    en  Led by Woody, Andy's toys live happily in his ...   
1                    en  When siblings Judy and Peter discover an encha...   
2                    en  A family wedding reignites the ancient feud be...   
3                    en  Cheated on, mistreated and stepped on, the wom...   
4                    en  Just when George Banks has recovered from his ...   
...                 ...                                                ...   
45460                en  Yet another version of the classic epic, with ...   
45462                tl  An artist struggles to finish his work while a...   
45463                en  When one of her hits goes wrong, a professiona...   
45464                en  In a small town live two brothers, one a minis...   
45465                en  50 years after decriminalisation of homosexual...   

      popularity                               production_companies  \
0      21.946943     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1      17.015539  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2        11.7129  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3       3.859495  [{'name': 'Twentieth Century Fox Film Corporat...   
4       8.387519  [{'name': 'Sandollar Productions', 'id': 5842}...   
...          ...                                                ...   
45460   5.683753  [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...   
45462   0.178241             [{'name': 'Sine Olivia', 'id': 19653}]   
45463   0.903007  [{'name': 'American World Pictures', 'id': 6165}]   
45464   0.003503               [{'name': 'Yermoliev', 'id': 88753}]   
45465   0.163015                                                 []   

                                    production_countries release_date  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   
...                                                  ...          ...   
45460  [{'iso_3166_1': 'CA', 'name': 'Canada'}, {'iso...   1991-05-13   
45462      [{'iso_3166_1': 'PH', 'name': 'Philippines'}]   2011-11-17   
45463  [{'iso_3166_1': 'US', 'name': 'United States o...   2003-08-01   
45464           [{'iso_3166_1': 'RU', 'name': 'Russia'}]   

In [65]:
#renombramos la nueva columna name por collection_name
df = df.rename(columns={'name': 'collection_name'})

In [66]:
#siguiendo la consigna iteramos para las columnas genres , production_companies , production_countries y spoken_languages

df['genres'] = df['genres'].apply(str_to_dict)

In [67]:
dfgenres = pd.json_normalize(df['genres'])

In [120]:
dfgenres

0                              1  \
0       {'id': 16, 'name': 'Animation'}   {'id': 35, 'name': 'Comedy'}   
1       {'id': 12, 'name': 'Adventure'}  {'id': 14, 'name': 'Fantasy'}   
2      {'id': 10749, 'name': 'Romance'}   {'id': 35, 'name': 'Comedy'}   
3          {'id': 35, 'name': 'Comedy'}    {'id': 18, 'name': 'Drama'}   
4          {'id': 35, 'name': 'Comedy'}                           None   
...                                 ...                            ...   
45459       {'id': 18, 'name': 'Drama'}   {'id': 28, 'name': 'Action'}   
45460       {'id': 18, 'name': 'Drama'}                           None   
45461      {'id': 28, 'name': 'Action'}    {'id': 18, 'name': 'Drama'}   
45462                              None                           None   
45463                              None                           None   

                                      2     3     4     5     6     7  
0       {'id': 10751, 'name': 'Family'}  None  None  None  None  None  
1       {'id': 10751, 'name': 'Family'}  None  None  None  None  None  
2                                  None  None  None  None  None  None  
3      {'id': 10749, 'name': 'Romance'}  None  None  None  None  None  
4                                  None  None  None  None  None  None  
...                                 ...   ...   ...   ...   ...   ...  
45459  {'id': 10749, 'name': 'Romance'}  None  None  None  None  None  
45460                              None  None  None  None  None  None  
45461    {'id': 53, 'name': 'Thriller'}  None  None  None  None  None  
45462                              None  None  None  None  None  None  
45463                              None  None  None  None  None  None  

[45464 rows x 8 columns]

In [104]:
#estudiamos las nuevas columnas generadas para mayor conocimiento

for i in dfgenres:
    print(f"cantidad de datos Nan de la columna {i}: {dfgenres[i].isna().value_counts()}")

cantidad de datos Nan de la columna 0: 0
False    42984
True      2480
Name: count, dtype: int64
cantidad de datos Nan de la columna 1: 1
False    28440
True     17024
Name: count, dtype: int64
cantidad de datos Nan de la columna 2: 2
True     31495
False    13969
Name: count, dtype: int64
cantidad de datos Nan de la columna 3: 3
True     41069
False     4395
Name: count, dtype: int64
cantidad de datos Nan de la columna 4: 4
True     44446
False     1018
Name: count, dtype: int64
cantidad de datos Nan de la columna 5: 5
True     45280
False      184
Name: count, dtype: int64
cantidad de datos Nan de la columna 6: 6
True     45437
False       27
Name: count, dtype: int64
cantidad de datos Nan de la columna 7: 7
True     45461
False        3
Name: count, dtype: int64


como se puede observar se generaron 8 columnas que contienen diccionarios, pero solo las primeras 2 tienen mas datos NotNAN que datos NAN, entonces nos quedamos con las columnas 0 y 1 descartando el resto

In [68]:
dfgenres = dfgenres.iloc[:, :2]

In [22]:
dfgenres

0                              1
0       {'id': 16, 'name': 'Animation'}   {'id': 35, 'name': 'Comedy'}
1       {'id': 12, 'name': 'Adventure'}  {'id': 14, 'name': 'Fantasy'}
2      {'id': 10749, 'name': 'Romance'}   {'id': 35, 'name': 'Comedy'}
3          {'id': 35, 'name': 'Comedy'}    {'id': 18, 'name': 'Drama'}
4          {'id': 35, 'name': 'Comedy'}                           None
...                                 ...                            ...
45459       {'id': 18, 'name': 'Drama'}   {'id': 28, 'name': 'Action'}
45460       {'id': 18, 'name': 'Drama'}                           None
45461      {'id': 28, 'name': 'Action'}    {'id': 18, 'name': 'Drama'}
45462                              None                           None
45463                              None                           None

[45464 rows x 2 columns]

In [69]:
dfgenres = pd.concat([pd.json_normalize(dfgenres[0]), dfgenres.drop([0], axis=1)], axis=1)

In [125]:
dfgenres

id       name                              1
0         16.0  Animation   {'id': 35, 'name': 'Comedy'}
1         12.0  Adventure  {'id': 14, 'name': 'Fantasy'}
2      10749.0    Romance   {'id': 35, 'name': 'Comedy'}
3         35.0     Comedy    {'id': 18, 'name': 'Drama'}
4         35.0     Comedy                           None
...        ...        ...                            ...
45459     18.0      Drama   {'id': 28, 'name': 'Action'}
45460     18.0      Drama                           None
45461     28.0     Action    {'id': 18, 'name': 'Drama'}
45462      NaN        NaN                           None
45463      NaN        NaN                           None

[45464 rows x 3 columns]

In [70]:
#para evitar conflictos con el desanidado de la columna 1 cambio el nombre de name y elimino ID por no ser relevante para el algoritmo ML
dfgenres = dfgenres.rename(columns={'name': 'genre_name0'})
dfgenres = dfgenres.drop ("id", axis=1)


desanido la columna 1 y repito con nombre name1

In [71]:
dfgenres = pd.concat([pd.json_normalize(dfgenres[1]), dfgenres.drop([1], axis=1)], axis=1)
dfgenres = dfgenres.rename(columns={'name': 'genre_name1'})
dfgenres = dfgenres.drop ("id", axis=1)

Concateno el dataframe original con la info de los generos

In [72]:
df_genres = pd.concat([df, dfgenres], axis =1)

In [73]:
#elimino la columna "genres" original anidada
df_genres = df_genres.drop ("genres", axis = 1)

In [144]:
df_genres

collection_name      budget original_language  \
0                Toy Story Collection  30000000.0                en   
1                                 NaN  65000000.0                en   
2           Grumpy Old Men Collection         0.0                en   
3                                 NaN  16000000.0                en   
4      Father of the Bride Collection         0.0                en   
...                               ...         ...               ...   
45463                             NaN         0.0                en   
45464                             NaN         0.0                en   
45465                             NaN         0.0                en   
45410                             NaN         NaN               NaN   
45461                             NaN         NaN               NaN   

      popularity                               production_companies  \
0      21.946943     [{'name': 'Pixar Animation Studios', 'id': 3}]   
1      17.015539  [{'name': 'TriStar Pictures', 'id': 559}, {'na...   
2        11.7129  [{'name': 'Warner Bros.', 'id': 6194}, {'name'...   
3       3.859495  [{'name': 'Twentieth Century Fox Film Corporat...   
4       8.387519  [{'name': 'Sandollar Productions', 'id': 5842}...   
...          ...                                                ...   
45463   0.903007  [{'name': 'American World Pictures', 'id': 6165}]   
45464   0.003503               [{'name': 'Yermoliev', 'id': 88753}]   
45465   0.163015                                                 []   
45410        NaN                                                NaN   
45461        NaN                                                NaN   

                                    production_countries release_date  \
0      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-10-30   
1      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-15   
2      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
3      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-12-22   
4      [{'iso_3166_1': 'US', 'name': 'United States o...   1995-02-10   
...                                                  ...          ...   
45463  [{'iso_3166_1': 'US', 'name': 'United States o...   2003-08-01   
45464           [{'iso_3166_1': 'RU', 'name': 'Russia'}]   1917-10-21   
45465   [{'iso_3166_1': 'GB', 'name': 'United Kingdom'}]   2017-06-09   
45410                                                NaN          NaT   
45461                                                NaN          NaT   

           revenue  runtime  \
0      373554033.0     81.0   
1      262797249.0    104.0   
2              0.0    101.0   
3       81452156.0    127.0   
4       76578911.0    106.0   
...            ...      ...   
45463          0.0     90.0   
45464          0.0     87.0   
45465          0.0     75.0   
45410          NaN      NaN   
45461          NaN      NaN   

                                        spoken_languages  \
0               [{'iso_639_1': 'en', 'name': 'English'}]   
1      [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2               [{'iso_639_1': 'en', 'name': 'English'}]   
3               [{'iso_639_1': 'en', 'name': 'English'}]   
4               [{'iso_639_1': 'en', 'name': 'English'}]   
...                                                  ...   
45463           [{'iso_639_1': 'en', 'name': 'English'}]   
45464                                                 []   
45465           [{'iso_639_1': 'en', 'name': 'English'}]   
45410                                                NaN   
45461                                                NaN   

                             title  vote_average  release_year     return  \
0                        Toy Story           7.7        1995.0  12.451801   
1                          Jumanji           6.9        1995.0   4.043035   
2                 Grumpier Old Men           6.5        1995.0        NaN   
3                Waiting to Exhale   

desanidado de la columna production_companies

In [74]:
#convierto a diccionario el tipo de dato de la columna
df_genres['production_companies'] = df_genres['production_companies'].apply(str_to_dict)

In [75]:
dfcompanies = pd.json_normalize(df_genres['production_companies'])

In [158]:
for i in dfcompanies:
    print(f"cantidad de datos Nan de la columna {i}: {dfcompanies[i].isna().sum()}")

cantidad de datos Nan de la columna 0: 11891
cantidad de datos Nan de la columna 1: 28463
cantidad de datos Nan de la columna 2: 36423
cantidad de datos Nan de la columna 3: 41139
cantidad de datos Nan de la columna 4: 43086
cantidad de datos Nan de la columna 5: 44035
cantidad de datos Nan de la columna 6: 44598
cantidad de datos Nan de la columna 7: 44912
cantidad de datos Nan de la columna 8: 45092
cantidad de datos Nan de la columna 9: 45225
cantidad de datos Nan de la columna 10: 45292
cantidad de datos Nan de la columna 11: 45334
cantidad de datos Nan de la columna 12: 45369
cantidad de datos Nan de la columna 13: 45388
cantidad de datos Nan de la columna 14: 45404
cantidad de datos Nan de la columna 15: 45413
cantidad de datos Nan de la columna 16: 45434
cantidad de datos Nan de la columna 17: 45441
cantidad de datos Nan de la columna 18: 45444
cantidad de datos Nan de la columna 19: 45449
cantidad de datos Nan de la columna 20: 45454
cantidad de datos Nan de la columna 21: 4545

visto la cantidad de valores NAN solo nos quedamos con la primer columna

In [76]:
dfcompanies = dfcompanies.iloc[:, :1]

In [77]:
#desanidamos
dfcompanies = pd.json_normalize(dfcompanies[0])
dfcompanies = dfcompanies.drop ("id", axis = 1)
dfcompanies = dfcompanies.rename(columns={'name': 'companie_name'})



Concateno con Df gral y elimino la columna "Production companie" original anidada

In [78]:
df_gcompanie = pd.concat([df_genres, dfcompanies], axis =1)

In [79]:
df_gcompanie = df_gcompanie.drop ("production_companies", axis = 1)

Desanido la columna "production_countries"

In [80]:
#convierto a diccionario el tipo de dato de la columna
df_gcompanie['production_countries'] = df_gcompanie['production_countries'].apply(str_to_dict)

In [81]:
#desanido
dfcountrie = pd.json_normalize(df_gcompanie['production_countries'])

In [46]:
for i in dfcountrie:
    print(f"cantidad de datos Nan de la columna {i}: {dfcountrie[i].isna().sum()}")

cantidad de datos Nan de la columna 0: 6308
cantidad de datos Nan de la columna 1: 38441
cantidad de datos Nan de la columna 2: 43315
cantidad de datos Nan de la columna 3: 44783
cantidad de datos Nan de la columna 4: 45245
cantidad de datos Nan de la columna 5: 45396
cantidad de datos Nan de la columna 6: 45435
cantidad de datos Nan de la columna 7: 45447
cantidad de datos Nan de la columna 8: 45454
cantidad de datos Nan de la columna 9: 45460
cantidad de datos Nan de la columna 10: 45461
cantidad de datos Nan de la columna 11: 45463
cantidad de datos Nan de la columna 12: 45464
cantidad de datos Nan de la columna 13: 45464
cantidad de datos Nan de la columna 14: 45464
cantidad de datos Nan de la columna 15: 45465
cantidad de datos Nan de la columna 16: 45465
cantidad de datos Nan de la columna 17: 45465
cantidad de datos Nan de la columna 18: 45465
cantidad de datos Nan de la columna 19: 45465
cantidad de datos Nan de la columna 20: 45465
cantidad de datos Nan de la columna 21: 45465

visto la cantidad de valores NAN solo nos quedamos con la primer columna

In [82]:
dfcountrie = dfcountrie.iloc[:, :1]

In [83]:
#desanidamos
dfcountrie = pd.json_normalize(dfcountrie[0])
dfcountrie = dfcountrie.drop ("iso_3166_1", axis = 1)
dfcountrie = dfcountrie.rename(columns={'name': 'country'})

Concateno con Df gral y elimino la columna "Production countrie" original anidada

In [84]:
df_gccountry = pd.concat([df_gcompanie, dfcountrie], axis =1)
df_gccountry= df_gccountry.drop("production_countries", axis = 1)

In [85]:
df_gccountry

collection_name      budget      id original_language  \
0                Toy Story Collection  30000000.0     862                en   
1                                 NaN  65000000.0    8844                en   
2           Grumpy Old Men Collection         0.0   15602                en   
3                                 NaN  16000000.0   31357                en   
4      Father of the Bride Collection         0.0   11862                en   
...                               ...         ...     ...               ...   
45463                             NaN         0.0   67758                en   
45464                             NaN         0.0  227506                en   
45465                             NaN         0.0  461257                en   
45410                             NaN         NaN     NaN               NaN   
45461                             NaN         NaN     NaN               NaN   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45463  When one of her hits goes wrong, a professiona...   0.903007   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   
45410                                                NaN        NaN   
45461                                                NaN        NaN   

      release_date      revenue  runtime  \
0       1995-10-30  373554033.0     81.0   
1       1995-12-15  262797249.0    104.0   
2       1995-12-22          0.0    101.0   
3       1995-12-22   81452156.0    127.0   
4       1995-02-10   76578911.0    106.0   
...            ...          ...      ...   
45463   2003-08-01          0.0     90.0   
45464   1917-10-21          0.0     87.0   
45465   2017-06-09          0.0     75.0   
45410          NaT          NaN      NaN   
45461          NaT          NaN      NaN   

                                        spoken_languages  \
0               [{'iso_639_1': 'en', 'name': 'English'}]   
1      [{'iso_639_1': 'en', 'name': 'English'}, {'iso...   
2               [{'iso_639_1': 'en', 'name': 'English'}]   
3               [{'iso_639_1': 'en', 'name': 'English'}]   
4               [{'iso_639_1': 'en', 'name': 'English'}]   
...                                                  ...   
45463           [{'iso_639_1': 'en', 'name': 'English'}]   
45464                                                 []   
45465           [{'iso_639_1': 'en', 'name': 'English'}]   
45410                                                NaN   
45461                                                NaN   

                             title  vote_average  vote_count  release_year  \
0                        Toy Story           7.7      5415.0        1995.0   
1                          Jumanji           6.9      2413.0        1995.0   
2                 Grumpier Old Men           6.5        92.0        1995.0   
3                Waiting to Exhale           6.1        34.0        1995.0   
4      Father of the Bride Part II           5.7       173.0        1995.0   
...                            ...           ...         ...           ...   
45463                     Betrayal           3.8         6.0        2003.0   
45464             Satan Triumphant           0.0         0.0        1917.0   
45465                     Queerama           0.0         0.0        2017.0   
45410                          NaN           NaN         NaN           NaN   
45461                          NaN           NaN         NaN           NaN   

          return genre_name1 genr

Desanidado de la columna "spoken_languages"

In [86]:
#convierto a diccionario el tipo de dato de la columna
df_gccountry['spoken_languages'] = df_gccountry['spoken_languages'].apply(str_to_dict)

In [87]:
#desanido
dflanguage = pd.json_normalize(df_gccountry['spoken_languages'])

In [195]:
for i in dflanguage:
    print(f"cantidad de datos Nan de la columna {i}: {dflanguage[i].isna().sum()}")

cantidad de datos Nan de la columna 0: 3866
cantidad de datos Nan de la columna 1: 37573
cantidad de datos Nan de la columna 2: 42943
cantidad de datos Nan de la columna 3: 44647
cantidad de datos Nan de la columna 4: 45197
cantidad de datos Nan de la columna 5: 45375
cantidad de datos Nan de la columna 6: 45437
cantidad de datos Nan de la columna 7: 45451
cantidad de datos Nan de la columna 8: 45457
cantidad de datos Nan de la columna 9: 45462
cantidad de datos Nan de la columna 10: 45463
cantidad de datos Nan de la columna 11: 45463
cantidad de datos Nan de la columna 12: 45464
cantidad de datos Nan de la columna 13: 45465
cantidad de datos Nan de la columna 14: 45465
cantidad de datos Nan de la columna 15: 45465
cantidad de datos Nan de la columna 16: 45465
cantidad de datos Nan de la columna 17: 45465
cantidad de datos Nan de la columna 18: 45465


Nuevamente debido a la cantidad de NAN presentes nos quedamos solo con la columna 1 descartando el resto

In [88]:
dflanguage = dflanguage.iloc[:, :1]

In [89]:
#desanidamos
dflanguage = pd.json_normalize(dflanguage[0])
dflanguage = dflanguage.drop ("iso_639_1", axis = 1)
dflanguage = dflanguage.rename(columns={'name': 'spoken'})



Concateno con Df gral y elimino la columna "spoken_languages" original anidada

In [90]:
df_final = pd.concat([df_gccountry, dflanguage], axis =1)
df_final= df_final.drop("spoken_languages", axis = 1)

In [91]:
df_final

collection_name      budget      id original_language  \
0                Toy Story Collection  30000000.0     862                en   
1                                 NaN  65000000.0    8844                en   
2           Grumpy Old Men Collection         0.0   15602                en   
3                                 NaN  16000000.0   31357                en   
4      Father of the Bride Collection         0.0   11862                en   
...                               ...         ...     ...               ...   
45463                             NaN         0.0   67758                en   
45464                             NaN         0.0  227506                en   
45465                             NaN         0.0  461257                en   
45410                             NaN         NaN     NaN               NaN   
45461                             NaN         NaN     NaN               NaN   

                                                overview popularity  \
0      Led by Woody, Andy's toys live happily in his ...  21.946943   
1      When siblings Judy and Peter discover an encha...  17.015539   
2      A family wedding reignites the ancient feud be...    11.7129   
3      Cheated on, mistreated and stepped on, the wom...   3.859495   
4      Just when George Banks has recovered from his ...   8.387519   
...                                                  ...        ...   
45463  When one of her hits goes wrong, a professiona...   0.903007   
45464  In a small town live two brothers, one a minis...   0.003503   
45465  50 years after decriminalisation of homosexual...   0.163015   
45410                                                NaN        NaN   
45461                                                NaN        NaN   

      release_date      revenue  runtime                        title  \
0       1995-10-30  373554033.0     81.0                    Toy Story   
1       1995-12-15  262797249.0    104.0                      Jumanji   
2       1995-12-22          0.0    101.0             Grumpier Old Men   
3       1995-12-22   81452156.0    127.0            Waiting to Exhale   
4       1995-02-10   76578911.0    106.0  Father of the Bride Part II   
...            ...          ...      ...                          ...   
45463   2003-08-01          0.0     90.0                     Betrayal   
45464   1917-10-21          0.0     87.0             Satan Triumphant   
45465   2017-06-09          0.0     75.0                     Queerama   
45410          NaT          NaN      NaN                          NaN   
45461          NaT          NaN      NaN                          NaN   

       vote_average  vote_count  release_year     return genre_name1  \
0               7.7      5415.0        1995.0  12.451801      Comedy   
1               6.9      2413.0        1995.0   4.043035     Fantasy   
2               6.5        92.0        1995.0        NaN      Comedy   
3               6.1        34.0        1995.0   5.090760       Drama   
4               5.7       173.0        1995.0        inf         NaN   
...             ...         ...           ...        ...         ...   
45463           3.8         6.0        2003.0        NaN         NaN   
45464           0.0         0.0        1917.0        NaN         NaN   
45465           0.0         0.0        2017.0        NaN         NaN   
45410           NaN         NaN           NaN        NaN       Drama   
45461           NaN         NaN           NaN        NaN       Drama   

      genre_name0                           companie_name  \
0       Animation                 Pixar Animation Studios   
1       Adventure                        TriStar Pictures   
2         Romance                            Warner Bros.   
3          Comedy  Twentieth Century Fox Film Corporation   
4          Comedy                   Sandollar Productions   
...           ...                                     ...   
45463         NaN                                  

In [92]:
#convertimos a float la columna popularity porque presentaba valores de tipo str y arrojaba error al generar archivo parquet
df_final['popularity'] = df_final['popularity'].astype(float)

In [94]:
df_final.to_parquet ("movies_etl.parquet")